test/Course/StateTest.hs

```diff
@@ -1,35 +1,51 @@
 {-# OPTIONS_GHC -fno-warn-type-defaults #-}
 {-# LANGUAGE NoImplicitPrelude   #-}
+{-# LANGUAGE OverloadedStrings   #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 
-module Course.StateTest where
+module Course.StateTest (
+  -- * Tests
+    test_State
+  , getTest
+  , putTest
+  , functorTest
+  , applicativeTest
+  , monadTest
+  , findMTest
+  , firstRepeatTest
+  , distinctTest
+  , isHappyTest
+  --   execTest
+  -- , evalTest
 
-import           Data.List                (nub)
-import qualified Prelude                  as P ((++))
+  -- * Course test runner
+  , courseTest
+  ) where
 
-import           Test.QuickCheck.Function (Fun (..))
-import           Test.Tasty               (TestTree, testGroup)
-import           Test.Tasty.HUnit         (testCase, (@?=))
-import           Test.Tasty.QuickCheck    (testProperty)
+import           Data.List          (nub)
+import qualified Prelude            as P ((++))
 
-import           Course.Applicative       (pure, (<*>))
+import           Course.Gens        (genInteger, genList)
+import           Test.Course.Mini   (courseTest)
+import           Test.Mini          (MiniTestTree, PropertyTester (..),
+                                     Tester (..), UnitTester (..), fn)
+
+import           Course.Applicative (pure, (<*>))
 import           Course.Core
-import           Course.Functor           ((<$>))
-import           Course.List              (List (..), filter, flatMap, hlist,
-                                           length, listh, span, (++))
-import           Course.Gens          (forAllLists)
+import           Course.Functor     ((<$>))
+import           Course.List        (List (..), filter, flatMap, hlist, length,
+                                     listh, span, (++))
 import           Course.Monad
-import           Course.Optional          (Optional (..))
-import           Course.State             (State (..), distinct, eval, exec,
-                                           findM, firstRepeat, get, isHappy,
-                                           put, put, runState)
+import           Course.Optional    (Optional (..))
+import           Course.State       (State (..), distinct, eval, exec, findM,
+                                     firstRepeat, get, isHappy, put, runState)
 
-test_State :: TestTree
+test_State :: MiniTestTree
 test_State =
   testGroup "State" [
-    execTest
-  , evalTest
-  , getTest
+  --   execTest
```

Comment:
Is the commented-out code in this file meant to be here?

Reply:
These two are properties that require arbitrary functions, which aren't currently implemented for the built-in tests. I was leaving them commented so they weren't forgotten. Happy to leave an explanatory comment and get rid of the commented code. Thoughts?